forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,404 @@
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use chrono::{DateTime, Utc};
+use clap::Subcommand;
+use colored::*;
+use forest_blocks::tipset_json::TipsetJson;
+use forest_cli_shared::cli::CliOpts;
+use forest_rpc_client::{
+    chain_get_name, chain_head, node_ops::node_status, start_time, wallet_balance,
+    wallet_default_address,
+};
+use forest_shim::econ::TokenAmount;
+use forest_utils::misc::LoggingColor;
+
+use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+use fvm_shared::{clock::ChainEpoch, BLOCKS_PER_EPOCH};
+use num::BigInt;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use super::Config;
+use crate::cli::handle_rpc_err;
+use crate::humantoken::TokenAmountPretty;
+
+#[derive(Debug, Subcommand)]
+pub enum InfoCommand {
+    Show,
+}
+
+#[derive(Debug)]
+pub struct NodeStatusInfo {
+    /// How far behind the node is with respect to syncing to head
+    pub behind: Duration,
+    /// Chain health is the percentage denoting how close we are to having
+    /// an average of 5 blocks per tipset in the last couple of
+    /// hours. The number of blocks per tipset is non-deterministic
+    /// but averaging at 5 is considered healthy.
+    pub health: f64,
+    /// epoch the node is currently at
+    pub epoch: ChainEpoch,
+    /// BaseFee is the set price per unit of gas (measured in attoFIL/gas unit) to be burned (sent to an unrecoverable address) for every message execution
+    pub base_fee: TokenAmount,
+    /// sync status information
+    pub sync_status: SyncStatus,
+    /// Start time of the node
+    pub start_time: DateTime<Utc>,
+    /// Current network the node is running on
+    pub network: String,
+    /// Default wallet address selected.
+    pub default_wallet_address: Option<String>,
+    /// Default wallet address balance
+    pub default_wallet_address_balance: Option<String>,
+}
+
+pub struct NodeInfoOutput {
+    pub chain_status: ColoredString,
+    pub network: ColoredString,
+    pub uptime: ColoredString,
+    pub health: ColoredString,
+    pub wallet_address: ColoredString,
+    pub wallet_balance: ColoredString,
+}
+
+#[derive(Debug, strum::Display, PartialEq)]
+pub enum SyncStatus {
+    Ok,
+    Slow,
+    Behind,
+}
+
+// Limits the format string from humantime::format_duration to exclude us ns ms parts.
+pub fn fmt_duration(duration: Duration) -> String {
+    let duration = humantime::format_duration(duration);
+    let duration = duration.to_string();
+    let duration = duration.split(' ');
+    let format_duration = duration
+        .filter(|s| !s.ends_with("us"))
+        .filter(|s| !s.ends_with("ns"))
+        .filter(|s| !s.ends_with("ms"))
+        .map(|s| s.to_string());
+    let format_duration: Vec<String> = format_duration.collect();
+    format_duration.join(" ")
+}
+
+impl NodeStatusInfo {
+    pub fn new(
+        cur_duration: Duration,
+        blocks_per_tipset_last_finality: f64,
+        head: TipsetJson,
+    ) -> anyhow::Result<NodeStatusInfo> {
```

Comment:
```suggestion
    ) -> Option<NodeStatusInfo> {
```

Not an appropriate error type here - there's only one reason this can fail, so document it.

Also please document what `blocks_per_tipset_last_finality` means, it's not clear from the name

Reply:
I agree, but i'd like to keep it this way for this instance, even though there's only one `Result` type here as it makes things a bit convenient within `run` function. Also, shouldn't we be reporting the error: `"System time should not be behind tipset timestamp, please sync the system clock."` to the user instead of just returning a `None`?

documented `blocks_per_tipset_last_finality` at usage.

Reply:
This function can no longer fail.